include/rmm/detail/memory_manager.hpp

```diff
@@ -31,223 +31,218 @@
-        }
-
-        /** ---------------------------------------------------------------------------*
-         * @brief Returns true if CUDA Managed Memory allocation is enabled
-         * 
-         * @return true if CUDA Managed Memory allocation is enabled
-         * @return false if CUDA Managed Memory allocation is disabled
-         * --------------------------------------------------------------------------**/
-        static inline bool useManagedMemory() {
-            return getOptions().allocation_mode & CudaManagedMemory;
-        }
-
-        /** ---------------------------------------------------------------------------*
-         * @brief Returns true when CUDA default allocation is enabled
-         *          * 
-         * @return true if CUDA default allocation is enabled
-         * @return false if CUDA default allocation is disabled
-         * --------------------------------------------------------------------------**/
-        inline bool useCudaDefaultAllocator() {
-            return CudaDefaultAllocation == getOptions().allocation_mode;
-        }
-
-        /** ---------------------------------------------------------------------------*
-         * @brief Register a new stream into the device memory manager.
-         * 
-         * Also returns success if the stream is already registered.
-         * 
-         * @param stream The stream to register
-         * @return rmmError_t RMM_SUCCESS if all goes well, RMM_ERROR_INVALID_ARGUMENT
-         *                    if the stream is invalid.
-         * ---------------------------------------------------------------------------**/
-        rmmError_t registerStream(cudaStream_t stream);
-
-    private:
-        Manager() : options({ CudaDefaultAllocation, false, 0 }), 
-                    is_initialized(false) {}
-        ~Manager() = default;
-        Manager(const Manager&) = delete;
-        Manager& operator=(const Manager&) = delete;
-  
-        std::mutex streams_mutex;
-        std::set<cudaStream_t> registered_streams;
-        Logger logger;
-
-        rmmOptions_t options;
-        bool is_initialized;
-    };    
+     * @param stream The stream to register
+     * @return rmmError_t RMM_SUCCESS if all goes well,
+     *                    RMM_ERROR_INVALID_ARGUMENT if the stream is invalid.
+     * ----------------------------------------------------------------------**/
+    rmmError_t registerStream(cudaStream_t stream);
+
+  private:
+    Manager() : options({ CudaDefaultAllocation, false, 0 }), 
+                is_initialized(false) {}
+    ~Manager() = default;
+    Manager(const Manager&) = delete;
+    Manager& operator=(const Manager&) = delete;
+    std::mutex manager_mutex;
+    std::set<cudaStream_t> registered_streams;
+    Logger logger;
+
+    rmmOptions_t options;
+    bool is_initialized;
+
+    // unique pointers to all allocated resources
+    std::unique_ptr<rmm::mr::device_memory_resource> memory_resource{};
```

Comment:
Consider renaming to make it more descriptive of what this is: it's a pointer to the resource that was created in initialization. 
```suggestion
    std::unique_ptr<rmm::mr::device_memory_resource> initialized_resource{};
```